nntrainer/models/neuralnet.cpp

```diff
@@ -277,6 +277,7 @@ int NeuralNetwork::initialize() {
  * @brief     free layers
  */
 NeuralNetwork::~NeuralNetwork() {
+  manager.reset();
```

Comment:
Was this causing the bug? manager should reset when NeuralNetwork destroys without this line.

Reply:
The bug was the missing check if the memory is already allocated.
This one just ensures that memory is released before layers are removed.